### PR TITLE
[incubator-kie-drools-6203] Adjust docs for 10.0.0

### PR DESCRIPTION
**Pull request by tkobayas**
**Issue**: 
* https://github.com/apache/incubator-kie-drools/issues/6203

    - Remove unused adoc files
    - Change kiegroup/drools to apache/incubator-kie-drools
    - Add entries to release notes

### CODE REVIEW

File: drools-docs/src/modules/ROOT/pages/release-notes/_release-notes-10.0.adoc [hunk 1]
```diff
@@ -19,6 +19,14 @@ with the License.  You may obtain a copy of the License at
...
\ No newline at end of file
+Quarkus Extension for Drools has been introduced. You can refer link:https://github.com/apache/incubator-kie-drools/blob/main/drools-quarkus-extension/drools-quarkus-quickstart-test/guide.adoc[Drools Quarkus extension Guide] for details.
+
+=== DRL on YAML
+
+As an experimental feature, DRL on YAML has been introduced, where you can write DRL in YAML format. You can refer link:https://github.com/apache/incubator-kie-drools/tree/main/drools-drlonyaml-parent[DRL on YAML] for details.
+
+=== Drools Reliability
+
+As an experimental feature, Drools Reliability has been introduced, which supports stateful KieSession persistence. You can refer link:https://github.com/apache/incubator-kie-drools/tree/main/drools-reliability[Drools Reliability] for details.
```
Comment (tkobayas) [inline drools-docs/src/modules/ROOT/pages/release-notes/_release-notes-10.0.adoc:43]: @mariofusco Do you agree with adding these 3 features in the release notes?
- Drools Quarkus Extension
- DRL on YAML (experimental)
- Drools Reliability (experimental)

Or do you prefer not to highlight some of the features in the release notes?

Also let me know if we have more topics to add in the release notes. ( https://github.com/apache/incubator-kie-drools/releases/tag/10.0.0 help us to recall what we did)

Comment (mariofusco) [reply drools-docs/src/modules/ROOT/pages/release-notes/_release-notes-10.0.adoc:43]: That's ok for me. I guess there are also some new DMN related features that we may want to add here, but I don't know the details. Maybe @gitgabrio and @yesamer could help with this?

**Review by mariofusco — APPROVED** [review]

**Comment by tkobayas** [comment]
@gitgabrio @yesamer @baldimir Let me know if you have anything to add in release notes. Of course, it could be added by another PR after merging this.

File: drools-docs/src/modules/ROOT/pages/release-notes/_release-notes-10.0.adoc (hunk 1 above)
Comment (yesamer) [reply drools-docs/src/modules/ROOT/pages/release-notes/_release-notes-10.0.adoc:43]: Hey @mariofusco @tkobayas! The main change on the DMN side is the support of DMN 1.5 specifications. You can add:
=== DMN 1.5 support
The DMN engine now supports DMN 1.5 specification. You can refer to the OMG official documentation to learn about the introduced features, like:
- Import into the Default Namespace
- New list replace() function
- Alternative image for input Data
- New type restriction attribute typeConstraint
- Possibility to negate a duration
- Possibility to iterate time ranges
- Date type to date and time type coercion

**Review by yesamer — APPROVED** [review]
Thank you @tkobayas

File: drools-docs/src/modules/ROOT/pages/release-notes/_release-notes-10.0.adoc (hunk 1 above)
Comment (tkobayas) [reply drools-docs/src/modules/ROOT/pages/release-notes/_release-notes-10.0.adoc:43]: Thanks @yesamer , that's a big update! Added on the top :)

**Pull request merged**